### PR TITLE
Remove dash dependency

### PR DESCRIPTION
**Pull request by areina**
Hello,

I created the PR with commits separated by function replacement because I think this way the testing and reviewing could be easier.  After the testing, if you prefer, I can squash the commits and generate only one with all changes. 

It isn't completely finished because I don't have a good replacement for `-some->>`, suggestions are welcome :-)

**update**: I consider it's finished

- [X] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [X] The new code is not generating byte compile warnings (run `cask exec emacs -batch -Q -L . -eval "(progn (setq byte-compile-error-on-warn t) (batch-byte-compile))" clj-refactor.el`)
- [X] All tests are passing (run `./run-tests.sh`)
- [X] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

**Comment by bbatsov** [comment]
The changes look good to me! Great work! 👍

**Comment by expez** [comment]
Thanks for taking this on!  Great work so far!

I'll give this a once over when you feel it's done, but it looks good so far!

@benedekfazekas are the hydra functions really unused, or are they used by the hydra package as an entry point into clj-refactor?

**Comment by benedekfazekas** [comment]
`hydra-docstring` and `hydra-heads` were used to generate the hydras for the first time based on `cljr--all-helpers`. No, they are not used directly by `hydra`. And as we are fine to manually update the `defhydra` macros I will remove them (easy to recover them from git history anyways...)

**Comment by expez** [comment]
@benedekfazekas they are removed in this pr, so they'll get removed when this is merged.  No need for you to do anything :)

**Comment by benedekfazekas** [comment]
whoops missed that, sry. already removed them.

**Comment by areina** [comment]
No worries, I rebased the PR with master. I think it's finished and ready for testing.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 - Remove cycle privacy, cycle if and cycle collection type features. They are moved to Clojure mode. The usual mnemonics for cycle privacy and cycle if features still work only they refer the Clojure mode implementations. Cycle collection type got reworked into convert collection to list, quoted list, map, vector and set. *Cycle* collection type is no longer supported.
 - Compatible with clojure-mode 5.6.1 and above.
 - Remove *most* of the code implementing `let` related refactorings because these featured got migrated and reworked in Clojure mode. The usual mnemonics for these features still work and also a version of these features using multiple cursors are still maintained here as multiple cursors mode is not supported in Clojure mode.
+- [#294] Remove dash library dependency.
```
Comment (expez) [inline CHANGELOG.md]: Does this short form actually create a link?

Comment (benedekfazekas) [inline CHANGELOG.md]: think so. you only need the long form afaik if the issue is in a different project

File: clj-refactor.el [hunk 2]
```diff
@@ -1282,31 +1284,34 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-move-form"
...
-                               (s-trim)))
+	      target-ns-alias (when-let ((filtered-require (seq-find
+							    (lambda (it)
+							      (s-matches-p (format target-ns-regexp-template ns) it))
+							    requires)))
+				(thread-last filtered-require
+				  (s-slice-at ":as")
+				  (last)
+				  (car)
+				  (replace-regexp-in-string (format target-ns-alias-template ns) "\\1")
```
Comment (expez) [inline clj-refactor.el]: Is this part of the thread able to continue if there is no alias and `nil` starts propagating through these forms?

**Comment by expez** [comment]
I've gone through it and apart from that one thing it looks great to me :)

Don't worry about squashing the commits, I can do that using the new squash-and-merge feature on github and I'd like to push one more commit at the end which resets the formatting (no tabs :p)

File: clj-refactor.el (hunk 2 above)
Comment (areina) [inline clj-refactor.el]: I would say this part of the thread always returns a string, so `nil` shouldn't be propagated. But this new expression is less flexible than former with `-some->>`, and if someone introduces some new function in the thread, should be more careful with it.

**Comment by areina** [comment]
I'm sorry about the tabs! I had something misconfigured. I pushed a new commit fixing it.

**Comment by expez** [comment]
Thanks for this massive patch, @areina! Outstanding work! :+1:

**Pull request merged**